migrations/2_deploy_contracts.js

```diff
@@ -53,6 +53,42 @@ module.exports = async function(deployer, network, accounts) {
     // deploy MarginVault library
     await deployer.deploy(MarginVault, {from: deployerAddress})
 
+    // deploy AddressBook & transfer ownership
+    await deployer.deploy(AddressBook, {from: deployerAddress})
+    const addressbook = await AddressBook.deployed()
+    // deploy OtokenFactory & set address
+    await deployer.deploy(OtokenFactory, addressbook.address, {from: deployerAddress})
+    const otokenFactory = await OtokenFactory.deployed()
+    await addressbook.setOtokenFactory(otokenFactory.address, {from: deployerAddress})
+    // deploy Otoken implementation & set address
+    await deployer.deploy(Otoken, {from: deployerAddress})
+    const otokenImpl = await Otoken.deployed()
+    await addressbook.setOtokenImpl(otokenImpl.address, {from: deployerAddress})
+    // deploy Whitelist module & set address
+    await deployer.deploy(Whitelist, addressbook.address, {from: deployerAddress})
+    const whitelist = await Whitelist.deployed()
+    await addressbook.setWhitelist(whitelist.address, {from: deployerAddress})
+    // deploy Oracle module & set address
+    await deployer.deploy(Oracle, {from : deployerAddress})
+    const oracle = await Oracle.deployed()
+    await addressbook.setOracle(oracle.address, {from: deployerAddress})
+    // deploy MarginPool module & set address
+    await deployer.deploy(MarginPool, addressbook.address, {from: deployerAddress})
+    const pool = await MarginPool.deployed()
+    await addressbook.setMarginPool(pool.address)
+    // deploy Calculator module & set address
```

Comment:
Isn't the deployment logic on mainnet and testnet the same? 
Why do we need the` if (network === "mainnet")` and `(network === 'testnet')` condition checks?

(This may not be related to this particular PR directly, just want to bring this up and see if we want to remove some duplicated logic)

Reply:
Yeah it is the same deployment logic, just wanted to separate those and make it more clean.

Reply:
why would separate them make it more clean? 😂

Reply:
No strong reason for tbh, I'm just used to do it that way. 
Can refactor them in one block in another PR.